exist-core/src/main/java/org/exist/xquery/Function.java

```diff
@@ -317,14 +317,13 @@ protected boolean checkArgumentTypeCardinality(final Expression argument, @Nulla
 
         if (!cardinalityMatches) {
             cardinalityMatches = argument.getCardinality().isSubCardinalityOrEqualOf(argType.getCardinality());
-            if (!cardinalityMatches) {
-                if (argument.getCardinality() == Cardinality.EMPTY_SEQUENCE
-                        && argType.getCardinality().atLeastOne()) {
-                    throw new XPathException(this,
-                            ErrorCodes.XPTY0004,
-                            Messages.getMessage(Error.FUNC_EMPTY_SEQ_DISALLOWED,
-                                    argPosition, ExpressionDumper.dump(argument)));
-                }
+        }
+
+        if (!cardinalityMatches) {
```

Comment:
cool, ... 😜 I would now combine the nested if-statements ... but that can also be addressed separately.